tests/test_ping_rx_basic/src/main.xc

```diff
@@ -1,174 +1,74 @@
-{
-    unsigned char buffer[1024];
-    unsigned char x;
-
-    for (int i = 0; i < length; i++)
-    {
-        buffer[i] = g_txDataCheck[epNum]++;
-
-        //asm("ld8u %0, %1[%2]":"=r"(x):"r"(g_txDataCheck),"r"(epNum));
-       // read_byte_via_xc_ptr_indexed(x, p_txDataCheck, epNum);
-
-        //buffer[i] = x;
-        //x++;
-        //asm("st8 %0, %1[%2]"::"r"(x),"r"(g_txDataCheck),"r"(epNum));
-        //write_byte_via_xc_ptr_indexed(p_txDataCheck,epNum,x);
-    }
-
-    XUD_SetBuffer(ep, buffer, length);
-}
-
-#pragma unsafe arrays
-int RxDataCheck(unsigned char b[], int l, int epNum)
-{
-    int fail = 0;
-    unsigned char x;
-
-    for (int i = 0; i < l; i++)
-    {
-        unsigned char y;
-        //read_byte_via_xc_ptr_indexed(y, p_rxDataCheck, epNum);
-        if(b[i] != g_rxDataCheck[epNum])
-        {
-            printstr("#### Mismatch on EP: ");
-            printint(epNum); 
-            printstr(". Got:");
-            printhex(b[i]);
-            printstr(" Expected:");
-            printhexln(g_rxDataCheck[epNum]);
-            //printintln(l); // Packet length
-            return 1;
-        }
-
-        g_rxDataCheck[epNum]++;
-        //read_byte_via_xc_ptr_indexed(x, p_rxDataCheck, epNum);
-        //x++;
-        //write_byte_via_xc_ptr_indexed(p_rxDataCheck,epNum,x);
-    }
-
-    return 0;
-}
-
-int TestEp_Bulk(chanend c_out1, chanend c_in1, int epNum1, chanend c_out2, chanend c_in2, int epNum2)
+int TestEp_PingTest(XUD_ep ep_out1, XUD_ep ep_out2, int epNum1, int epNum2)
 {
     unsigned int length;
-    XUD_Result_t res;
-
-    XUD_ep ep_out1 = XUD_InitEp(c_out1);
-    XUD_ep ep_out2 = XUD_InitEp(c_out2);
-    XUD_ep ep_in1  = XUD_InitEp(c_in1);
-    XUD_ep ep_in2 = XUD_InitEp(c_in2);
 
-    /* Buffer for Setup data */
     unsigned char buffer[1024];
 
     int i = 10;
-    {    
-        XUD_GetBuffer(ep_out1, buffer, length);
-
-        if(length != i)
-        {
-            printintln(length);
-            fail(FAIL_RX_LENERROR);
-        }
-
-        if(RxDataCheck(buffer, length, epNum1))
-        {
-            fail(FAIL_RX_DATAERROR);
-        }
+    
+    XUD_GetBuffer(ep_out1, buffer, length);
 
+    if(RxDataCheck(buffer, length, epNum1, i))
+    {
+        return FAIL_RX_DATAERROR;
     }
-
         
-    XUD_GetBuffer(ep_out2, buffer, length);
-    
-     if(length != i)
-        {
-            printintln(length);
-            fail(FAIL_RX_LENERROR);
-        }
+	XUD_GetBuffer(ep_out2, buffer, length);
 
-        if(RxDataCheck(buffer, length, epNum2))
-        {
-            fail(FAIL_RX_DATAERROR);
-        }
+	if(RxDataCheck(buffer, length, epNum2, i))
+	{
+		return FAIL_RX_DATAERROR;
+	}
 
+    // Another packet to "ctrl" EP means we can exit
+    XUD_GetBuffer(ep_out1, buffer, length);
 
-    exit(0);
+    if(RxDataCheck(buffer, length, epNum1, i))
+    {
+        return FAIL_RX_DATAERROR;
+    }
+	
+	return 0;
 }
 
+#define CTRL_EP_NUM (TEST_EP_NUM + 1)
 
```

Comment:
Quite a few main.xc files `#include "test_main.xc"` and use its common functions.  This file does not.  Is that intentional or an oversight?